source/neuropod/internal/neuropod_tensor.hh

```diff
@@ -338,6 +344,54 @@ public:
         return TensorAccessor<const T *, N>(get_raw_data_ptr(), get_dims().data(), get_strides().data());
     }
 
+    // Return a view of this tensor with the requested dimensions.
+    // `requested_dims` must be compatible with the dims of this tensor.
+    // This method does not make a copy of the tensor and changes made in the
+    // returned view are visible in the original tensor.
+    template <typename... Dim>
+    auto view(Dim... requested_dims)
+    {
+        this->assure_device_cpu();
+        std::vector<int64_t> new_dims = {requested_dims...};
+
+        // Check that the requested dims are compatible with this tensor
+        assure_view_compatible_shape(new_dims);
+
+        // Create and return a tensor view
+        constexpr size_t num_dims    = sizeof...(requested_dims);
+        auto             new_strides = detail::compute_strides(new_dims);
+        return TensorView<T *, num_dims>(get_raw_data_ptr(), std::move(new_dims), std::move(new_strides));
+    }
+
+    template <typename... Dim>
+    auto view(Dim... requested_dims) const
```

Comment:
I see const/non-const are different in last line, T/"const T'. I think it won't be hard to move it to private helper template method "view_type" that accepts type and then can be used by both views.